packages/web-app-files/src/mixins/actions/restore.ts

```diff
@@ -47,7 +57,89 @@ export default {
     ...mapMutations('runtime/spaces', ['UPDATE_SPACE_FIELD']),
     ...mapMutations(['SET_QUOTA']),
 
-    async $_restore_trigger({ resources }) {
+    getParentFolderFromResource(resource) {
+      const parentPath = resource.path.slice(0, resource.path.lastIndexOf(resource.name))
+      return parentPath
+    },
+    getWebdavParentFolderFromResource(resource) {
+      const parentPath = this.getParentFolderFromResource(resource)
+      return buildWebDavFilesPath(this.user.id, parentPath)
+    },
+    async collectRestoreConflicts(resources) {
+      const parentFolders = {}
+      const conflicts = []
+      const resolvedResources = []
+      for (const resource of resources) {
+        const webDavParentPath = this.getWebdavParentFolderFromResource(resource)
+
+        // ? check if parent folder has already been requested
+        let parentResources = []
+        if (webDavParentPath in parentFolders) {
+          parentResources = parentFolders[webDavParentPath]
+        } else {
+          const listResponse = await this.$client.files.list(
```

Comment:
This returns the root directory as well. I think you need to respect that in your check down below: `parentResources.some((e) => e.name === resource.name)`. Otherwise you could not restore a folder `aaa` inside folder `aaa`.

Not 100% sure though as I can't try it myself.

Reply:
yes you are right indeed